stitches/dev_matchup.py

```diff
@@ -0,0 +1,155 @@
+    :param model_name:         A str name of the model to emulate
+    :type data:                str
+    :param df:                 A data frame of the tgav data.
+    :type df:                  pandas data frame
+
+    :return:                   A pandas data frame of cmip tgav for a single model
+
+    # TODO hate the function name, wanted something that was easier to work with
+    """
+
+    if not (isinstance(df, pd.DataFrame)):
+        raise TypeError(f"df not imported")
+
+    if not (df.size >= 10):
+        raise TypeError(f"df is empty")
+
+    # Check to make sure that the model name is within the data frame.
+    if not (model_name in set(df["model"].unique())):
+        raise TypeError(f"model_name tgav data not avaiable")
+
+    # Now subset the data for a single model to emulate.
+    out = df[df["model"] == model_name].copy()
+    return out
+
+#########################################################################################################
+def calculate_anomaly(data):
+
+    """Convert the temp data from absolute into an anomaly
+    :param data:               A data frame of the cmip absolute temperature
+    :type data:                pandas.core.frame.DataFrame
+
+    :return:                   A pandas data frame of cmip temperature as tgav
+    """
+    # inputs
+    req_cols = {'activity', 'model', 'experiment', 'ensemble', 'timestep', 'grid_type',
+                'file', 'year', 'value', 'variable'}
+    col_names = set(data.columns)
+    if not (req_cols.issubset(col_names)):
+        raise TypeError(f'Missing columns from "{data}".')
+
+    to_use = data[['model', 'experiment', 'ensemble', 'year', 'value']].copy()
+    # Subset the data so that it only contains the values from our defined reference years
+    # we follow the IPPC/CMIP6 reference years 1995 to 2014.
+    startYr = 1995
+    endYr = 2014
+    subset_data = to_use[to_use['year'].between(startYr, endYr)]
+
+    # Calculate the mean value for each ensemble member, this will be used as the reference value that is used to
+    # convert fom absolute to relative temp.
+    reference_values = subset_data.groupby(['model', 'ensemble']).agg(
+        {'value': lambda x: sum(x) / len(x)}).reset_index().copy()
+    reference_values = reference_values.rename(columns={"value": "ref_values"})
+
+    # Combine the dfs that contain the absolute temperature values with the reference values.
+    merged_df = data.merge(reference_values, on=['model', 'ensemble'], how='left')
+    merged_df['value'] = merged_df['value'] - merged_df['ref_values']
+    merged_df = merged_df.drop(columns='ref_values')
+
+    # Return the data frame, note might want to sort so that the time series are in order but
+    # not really necessary.
+    return merged_df
+
+#########################################################################################################
+
+def calculate_rolling_mean(input_data, size):
+    """"
+    Calculate the rolling mean for the data frame
```

Comment:
Can we note if the mean being calculated on Line 153 is centered on each year, starting from each year, or ending on each year? In other words, is 2055 the average of 2052-2057,  2051-2055, or 2055-2059 if `size=5`?

Reply:
Looking at https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rolling.html which is the function we use to calculate the rolling mean right now it is setup to start the rolling on the right, do we want to do that or should we use the center mean? Also if the data does not meet the window size do we want it discarded? we could adjust it so if it isn't the right size we allow it to fewer values in the mean. 

@abigailsnyder and @claudiatebaldi 

I'm inclined to think that we should use the center rolling average instead of left or right orientated.